core/src/main/scala/org/bykn/bosatsu/Declaration.scala

```diff
@@ -598,7 +598,7 @@ object Declaration {
    * we also parse Bind, Def, Comment
    */
   private[this] val parserCache: ((ParseMode, String)) => P[Declaration] =
-    Memoize.function[(ParseMode, String), P[Declaration]] { case ((pm, indent), rec) =>
+    Memoize.memoizeDagHashed[(ParseMode, String), P[Declaration]] { case ((pm, indent), rec) =>
```

Comment:
This has to be thread safe because the parseCache is a static value and exposed by the Indy.

(bugs with this can be seen my doing parsing in parallel, which isn’t really needed).